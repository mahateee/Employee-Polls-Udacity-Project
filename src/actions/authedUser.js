export const SET_AUTHED_USER = "SET_AUTHED_USER ";
export function setAuthedUser(authedUser) {
  return {
    type: SET_AUTHED_USER,
    authedUser,
  };
}

export function handleLogin(username, password) {
  return (dispatch, getState) => {
    const { users } = getState();

    const user = Object.values(users).find(
      (user) => user.id === username && user.password === password
    );

    if (!!user) {
      return dispatch(setAuthedUser(user));
    }
  };
}
