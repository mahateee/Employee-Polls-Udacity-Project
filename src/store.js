import { legacy_createStore as createStore } from "redux";
import { rootReducer } from "./reducers/index";
import middleware from "./middleware";

export const store = createStore(rootReducer, middleware);
