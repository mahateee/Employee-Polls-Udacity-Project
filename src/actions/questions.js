import { _saveQuestion } from "../utils/_DATA";
import { saveQuestionAnswer, saveQuestion } from "../utils/api";
import { addAnswerToUser, addQuestionToUser } from "./users";
export const ADD_QUESTION = "ADD_QUESTION";
export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const ADD_ANSWER_TO_QUESTION = "ADD_ANSWER_TO_QUESTION";
export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}
export function addAnswerToQuestion({ authedUser, qid, answer }) {
  return {
    type: ADD_ANSWER_TO_QUESTION,
    authedUser,
    qid,
    answer,
  };
}

function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}
export function handleAddAnswer(info) {
  return (dispatch) => {
    saveQuestionAnswer(info)
      .then(() => {
        dispatch(addAnswerToQuestion(info));
        dispatch(addAnswerToUser(info));
      })
      .catch((e) => {
        console.warn("Error in handleToggleTweet: ", e);
        dispatch(addAnswerToQuestion(info));
        dispatch(addAnswerToUser(info));

        alert("There was an error liking the tweet. Try again.");
      });
  };
}
export function handleSaveQuestion(info) {
  return (dispatch) => {
    return saveQuestion(info).then((info) => {
      dispatch(addQuestion(info));
      dispatch(addQuestionToUser(info));
    });
  };
}
