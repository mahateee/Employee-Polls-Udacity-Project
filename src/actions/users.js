import { saveQuestionAnswer } from "../utils/api";
import { addAnswerToQuestion } from "./questions";
export const RECEIVE_USERS = "RECEIVE_USERS";
export const ADD_ANSWER_TO_USER = "ADD_ANSWER_TO_USER";
export const ADD_USER_QUESTION = "ADD_QUESTION_TO_USER";
export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function addAnswerToUser({ authedUser, qid, answer }) {
  return {
    type: ADD_ANSWER_TO_USER,
    authedUser,
    qid,
    answer,
  };
}
export function addQuestionToUser({ author, id }) {
  return {
    type: ADD_USER_QUESTION,
    author,
    qid: id,
  };
}
