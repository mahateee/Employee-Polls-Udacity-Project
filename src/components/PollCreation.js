import React from "react";
import { useState } from "react";
import { connect } from "react-redux";
import { handleSaveQuestion } from "../actions/questions";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
function PollCreation(props) {
  const { authedUser } = props;
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(
      handleSaveQuestion({
        optionOneText: firstOption,
        optionTwoText: secondOption,
        author: authedUser.id,
      })
    );
    navigate("/Dashboard");
  };
  return (
    <section class="bg-gray-50 dark:bg-gray-900">
      <div class="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0">
        <div class="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700">
          <div class="p-6 space-y-4 md:space-y-4 sm:p-8">
            <h1 class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white">
              would you rether
            </h1>
            <p class=" font-light text-gray-500 sm:text-xl dark:text-gray-400">
              create your own poll
            </p>
            <form
              onSubmit={handleSubmit}
              class="space-y-6 md:space-y-6"
              action="#"
            >
              <div>
                <label
                  for="option1"
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                >
                  option 1
                </label>
                <input
                  type="text"
                  name="option1"
                  id="option1"
                  value={firstOption}
                  onChange={(e) => setFirstOption(e.target.value)}
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  placeholder="option 1"
                  required=""
                />
              </div>
              <div>
                <label
                  for="option2"
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                >
                  option 2
                </label>
                <input
                  type="text"
                  value={secondOption}
                  onChange={(e) => setSecondOption(e.target.value)}
                  name="option2"
                  id="option2"
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  placeholder="option 2"
                  required=""
                />
              </div>
              <button class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">
                Read more
              </button>
            </form>
          </div>
        </div>
      </div>
    </section>
  );
}
function mapStateToProps({ authedUser }) {
  return {
    authedUser,
  };
}

export default connect(mapStateToProps)(PollCreation);
