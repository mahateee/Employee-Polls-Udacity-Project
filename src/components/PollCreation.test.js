import { render, screen, fireEvent } from "@testing-library/react";
import * as React from "react";
import PollCreation from "./PollCreation";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";
import { handleInitialData } from "../actions/shared";
describe("PollCreation", () => {
  it("should render the component", () => {
    store.dispatch(setAuthedUser({ id: "sarahedo" }));
    store.dispatch(handleInitialData());
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollCreation />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });
  it("should display all elements", () => {
    store.dispatch(setAuthedUser({ id: "sarahedo" }));
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollCreation />
        </BrowserRouter>
      </Provider>
    );
    const inputOne = screen.getByTestId("firstOption");
    const inputTwo = screen.getByTestId("secondOption");
    const submitButton = screen.getByTestId("submit-button");

    fireEvent.change(inputOne, { target: { value: "first value" } });
    fireEvent.change(inputTwo, { target: { value: "second value" } });
    expect(inputOne.value).toBe("first value");
    expect(inputTwo.value).toBe("second value");
  });
  it("will display an error if the firstOption & secondOption is not provided.", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollCreation />
        </BrowserRouter>
      </Provider>
    );

    const submitButton = screen.getByTestId("submit-button");
    fireEvent.click(submitButton);
    expect(screen.getByTestId("error-header")).toBeInTheDocument();
    expect(screen.queryByTestId("success-header")).not.toBeInTheDocument();
  });
  it("will display a success message if the firstOption & secondOption are provided.", () => {
    store.dispatch(setAuthedUser({ id: "sarahedo" }));
    store.dispatch(setAuthedUser({ id: "sarahedo" }));
    store.dispatch(handleInitialData());
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollCreation />
        </BrowserRouter>
      </Provider>
    );
    const inputOne = screen.getByTestId("firstOption");
    const inputTwo = screen.getByTestId("secondOption");
    const submitButton = screen.getByTestId("submit-button");

    fireEvent.change(inputOne, { target: { value: "first value" } });
    fireEvent.change(inputTwo, { target: { value: "second value" } });
    fireEvent.click(submitButton);
    expect(screen.getByTestId("success-header")).toBeInTheDocument();
    expect(screen.queryByTestId("error-header")).not.toBeInTheDocument();
  });
});
