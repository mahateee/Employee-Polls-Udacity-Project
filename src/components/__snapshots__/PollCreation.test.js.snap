// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollCreation should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="bg-gray-50 dark:bg-gray-900"
      >
        <header>
          <nav
            class="bg-white border-gray-200 px-4 lg:px-6 py-2.5 dark:bg-gray-800"
          >
            <div
              class="flex flex-wrap justify-between items-center mx-auto max-w-screen-xl"
            >
              <div
                class="flex items-center lg:order-2"
              >
                <div
                  class="text-gray-800 dark:text-white hover:bg-gray-50 focus:ring-4 focus:ring-gray-300 font-medium rounded-lg text-sm px-4 lg:px-5 py-2 lg:py-2.5 mr-2 dark:hover:bg-gray-700 focus:outline-none dark:focus:ring-gray-800"
                >
                  <div
                    data-testid="user-info"
                  />
                </div>
                <button
                  class="text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:ring-primary-300 font-medium rounded-lg text-sm px-4 lg:px-5 py-2 lg:py-2.5 mr-2 dark:bg-primary-600 dark:hover:bg-primary-700 focus:outline-none dark:focus:ring-primary-800"
                >
                  Logout
                </button>
                <button
                  aria-controls="mobile-menu-2"
                  aria-expanded="false"
                  class="inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg lg:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
                  data-collapse-toggle="mobile-menu-2"
                  type="button"
                >
                  <span
                    class="sr-only"
                  >
                    Open main menu
                  </span>
                  <svg
                    class="w-6 h-6"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    class="hidden w-6 h-6"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="hidden justify-between items-center w-full lg:flex lg:w-auto lg:order-1"
                id="mobile-menu-2"
              >
                <ul
                  class="flex flex-col mt-4 font-medium lg:flex-row lg:space-x-8 lg:mt-0"
                >
                  <li
                    class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                  >
                    <a
                      href="/Dashboard"
                    >
                      Dashboard
                    </a>
                  </li>
                  <li
                    class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                  >
                    <a
                      href="/Leaderboard"
                    >
                      Leaderboard
                    </a>
                  </li>
                  <li
                    class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                  >
                    <a
                      href="/New"
                    >
                      New
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </header>
        <div
          class="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
        >
          <div
            class="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
          >
            <div
              class="p-6 space-y-4 md:space-y-4 sm:p-8"
            >
              <h1
                class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
              >
                would you rether
              </h1>
              <p
                class=" font-light text-gray-500 sm:text-xl dark:text-gray-400"
              >
                create your own poll
              </p>
              <form
                action="#"
                class="space-y-6 md:space-y-6"
              >
                <div>
                  <label
                    class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                    for="option1"
                  >
                    option 1
                  </label>
                  <input
                    class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                    data-testid="firstOption"
                    id="option1"
                    name="option1"
                    placeholder="option 1"
                    type="text"
                    value=""
                  />
                </div>
                <div>
                  <label
                    class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                    for="option2"
                  >
                    option 2
                  </label>
                  <input
                    class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                    data-testid="secondOption"
                    id="option2"
                    name="option2"
                    placeholder="option 2"
                    type="text"
                    value=""
                  />
                </div>
                <button
                  class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                  data-testid="submit-button"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="bg-gray-50 dark:bg-gray-900"
    >
      <header>
        <nav
          class="bg-white border-gray-200 px-4 lg:px-6 py-2.5 dark:bg-gray-800"
        >
          <div
            class="flex flex-wrap justify-between items-center mx-auto max-w-screen-xl"
          >
            <div
              class="flex items-center lg:order-2"
            >
              <div
                class="text-gray-800 dark:text-white hover:bg-gray-50 focus:ring-4 focus:ring-gray-300 font-medium rounded-lg text-sm px-4 lg:px-5 py-2 lg:py-2.5 mr-2 dark:hover:bg-gray-700 focus:outline-none dark:focus:ring-gray-800"
              >
                <div
                  data-testid="user-info"
                />
              </div>
              <button
                class="text-white bg-primary-700 hover:bg-primary-800 focus:ring-4 focus:ring-primary-300 font-medium rounded-lg text-sm px-4 lg:px-5 py-2 lg:py-2.5 mr-2 dark:bg-primary-600 dark:hover:bg-primary-700 focus:outline-none dark:focus:ring-primary-800"
              >
                Logout
              </button>
              <button
                aria-controls="mobile-menu-2"
                aria-expanded="false"
                class="inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg lg:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
                data-collapse-toggle="mobile-menu-2"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open main menu
                </span>
                <svg
                  class="w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
                <svg
                  class="hidden w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden justify-between items-center w-full lg:flex lg:w-auto lg:order-1"
              id="mobile-menu-2"
            >
              <ul
                class="flex flex-col mt-4 font-medium lg:flex-row lg:space-x-8 lg:mt-0"
              >
                <li
                  class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                >
                  <a
                    href="/Dashboard"
                  >
                    Dashboard
                  </a>
                </li>
                <li
                  class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                >
                  <a
                    href="/Leaderboard"
                  >
                    Leaderboard
                  </a>
                </li>
                <li
                  class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-primary-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700"
                >
                  <a
                    href="/New"
                  >
                    New
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </header>
      <div
        class="flex flex-col items-center justify-center px-6 py-8 mx-auto md:h-screen lg:py-0"
      >
        <div
          class="w-full bg-white rounded-lg shadow dark:border md:mt-0 sm:max-w-md xl:p-0 dark:bg-gray-800 dark:border-gray-700"
        >
          <div
            class="p-6 space-y-4 md:space-y-4 sm:p-8"
          >
            <h1
              class="text-xl font-bold leading-tight tracking-tight text-gray-900 md:text-2xl dark:text-white"
            >
              would you rether
            </h1>
            <p
              class=" font-light text-gray-500 sm:text-xl dark:text-gray-400"
            >
              create your own poll
            </p>
            <form
              action="#"
              class="space-y-6 md:space-y-6"
            >
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                  for="option1"
                >
                  option 1
                </label>
                <input
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  data-testid="firstOption"
                  id="option1"
                  name="option1"
                  placeholder="option 1"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
                  for="option2"
                >
                  option 2
                </label>
                <input
                  class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
                  data-testid="secondOption"
                  id="option2"
                  name="option2"
                  placeholder="option 2"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="inline-flex items-center px-3 py-2 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                data-testid="submit-button"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
